<launch>
  <arg name="map_file"       default="$(find cse_190_assi_2)/map/stage.yaml"/>
  <arg name="world_file"     default="$(find cse_190_assi_2)/map/stage.world"/>

  <node pkg="cse_190_assi_2" type="robot.py" name="particle_filter_localization" output="screen" required="true"/> 
  <node pkg="stage_ros" type="stageros" name="stageros" args="$(arg world_file)">
    <param name="base_watchdog_timeout" value="1000"/>
    <remap from="base_scan" to="original_scan"/>
  </node>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="/map"/>
  </node>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find cse_190_assi_2)/rviz/PA2.rviz"/>
  <node pkg="cse_190_assi_2" type="laser_model.py" name="laser_model" output="screen" cwd="node"/> 
  <node pkg="cse_190_assi_2" type="test_metric.py" name="test_metric" output="screen" cwd="node"/> 

</launch>
